### ISSUE TITLE
Missing Clan Members from dashboard

### ISSUE THREAD

**Issue by madwed** [Codewars/codewars.com#109]
Some clan members are not showing up in the allies field of the dashboard.

**Comment by tansaku** [comment]
we have this too.  Our clan is 'Makers Academy', and I note that for example http://www.codewars.com/users/ljones140 has that clan, but does not show up there

**Comment by tansaku** [comment]
@jhoffner any chance this might be looked at soon?

**Comment by kmalin** [comment]
This is still an issue. I cannot see some clan members, and that person cannot see me, even though we are both in the same clan.

**Comment by jhoffner** [comment]
@kmalin did you follow each other? The Clan functionality basically just auto-follows users for you so you can follow them manually if need be. 

@tansaku I'm in the middle of big push for another project but I'll take a look ASAP.

**Comment by kmalin** [comment]
@jhoffner I haven't, previously. But now I tried to test it and I clicked Follow button on his page. I see that now the page shows a button Unfollow, but I still cannot see him on my dashboard page. Ah, perhaps he also needs to follow me? Relevant users are kabanga (me) and giedriust, if that helps.

**Comment by mcelearr** [comment]
@jhoffner Having the same issue with the 'Founders & Coders' clan. I'm not able to see certain clan members and they're not able to see me. But we're able to see 75% of the other clan members okay.

**Comment by marythought** [comment]
This is an issue with my code school, too. Clan Code Fellows.

**Comment by znalbert** [comment]
Just came here to see if this issue existed.  I tried following one of the clan members missing from my allies list and he still doesn't show up just like @kmalin's issue. My username on codewars is znalbert and his is forbiddenvoid. Also, czeise shows up on my allies, but I don't show up on his.

**Comment by camerican** [comment]
Not ideal, but you can re-save the clan settings of your profile to add missing clan members to your follow feed.  Encouraged to see that @jhoffner has added it to the official to-do list.

From the outside looking in, it seems when you join a clan, Codewars will have your account follow all users within the clan you subscribe to, but not have everybody in the clan reverse-follow you.  If this is the case, changing from a O(n) to O(n^2) write operation will become expensive for large clans, so it may be worthwhile to either defer new follows until a user logs back in or to have your feed just pull from a clan association table.

**Comment by ctscodes** [comment]
I can't see my own name in my clans dashboard. Is this the same issue or a new issue?

**Comment by Jonathan-Ngondi** [comment]
I also can't see my own name on my clan's dashboard. Was this issue resolved?

**Comment by danders-son** [comment]
Problem for me too.
Using my clans name has no effect in autofollowing. Tried to update the profile but this has no effect.

**Comment by xqwtsz** [comment]
clan members of the same name clan not showing up on "Allies" tab.

**Comment by VanessaGreene** [comment]
@kazk has this been resolved?

**Comment by kazk** [comment]
No. The name of this issue is also a bit misleading because the users listed on the dashboard are not just clan members, but allies (anyone following you back).
